### PR TITLE
Ibp 4440 move attributes to ontology

### PR DESCRIPTION
**Pull request by clarysabel**

### CODE REVIEW

File: src/main/jhipster/src/main/webapp/app/germplasm-manager/germplasm-import-update-dialog.component.ts [hunk 1]
```diff
@@ -53,18 +63,23 @@ export class GermplasmImportUpdateDialogComponent implements OnInit, OnDestroy {
...
+        try {
+            await this.confirmAttributeStatus();
+        } catch (e) {
+            this.isProcessing = false;
+            return;
+        }
+        this.germplasmService.importGermplasmUpdates(this.transform(this.data, this.names, this.attributes)).subscribe(
+            (res: HttpResponse<any>) => this.onSaveSuccess(res.body),
+            (res: HttpErrorResponse) => this.onError(res)
+        );
```
Comment (nahuel-soldevilla) [inline src/main/jhipster/src/main/webapp/app/germplasm-manager/germplasm-import-update-dialog.component.ts:82]: @abatac I refactored the  germplasm-import-update-dialog.component `import()` and
`validate()` methods using async/await syntax, to simplify adding more asynchronous confirmation steps. Let me know if you have any concerns

File: src/main/jhipster/src/main/webapp/app/shared/germplasm/service/germplasm.service.ts [hunk 2]
```diff
@@ -71,20 +71,15 @@ export class GermplasmService {
             { params, observe: 'response' });
     }
 
+    // FIXME into IBP-4659 (Replace by Resource /variables/filter)
```
Comment (clarysabel) [inline src/main/jhipster/src/main/webapp/app/shared/germplasm/service/germplasm.service.ts:74]: this function getGermplasmAttributes can be removed 
@cuenyad @nahuel-soldevilla

Comment (clarysabel) [reply src/main/jhipster/src/main/webapp/app/shared/germplasm/service/germplasm.service.ts:74]: done

File: src/main/jhipster/src/main/webapp/app/entities/germplasm/attribute/germplasm-attribute-modal.component.ts [hunk 3]
```diff
@@ -21,32 +27,40 @@ export class GermplasmAttributeModalComponent implements OnInit, OnDestroy {
     attributeCodes: Promise<Attribute[]>;
```
Comment (darla-leafnode) [inline src/main/jhipster/src/main/webapp/app/entities/germplasm/attribute/germplasm-attribute-modal.component.ts:27]: `attributeCodes` is now unused?

File: src/main/jhipster/src/main/webapp/app/shared/ontology/variable-type.enum.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+export enum VariableTypeEnum {
```
Comment (darla-leafnode) [inline src/main/jhipster/src/main/webapp/app/shared/ontology/variable-type.enum.ts:1]: I suggest just use `VariableType` (omit the enum)

File: src/main/jhipster/src/main/webapp/app/shared/variable-select/variable-select.component.ts [hunk 5]
```diff
@@ -13,8 +14,17 @@ declare const $: any;
...
-    @Input() variableTypeIds: string[];
+    @Input() name: string;
+    @Input() id: string;
+
+    @Input() value: any;
+    initialData: Select2OptionData[];
+
+    @Input() disabled = false;
+    @Input() multiple = false;
+    @Input() variableTypeIds: number[];
```
Comment (darla-leafnode) [inline src/main/jhipster/src/main/webapp/app/shared/variable-select/variable-select.component.ts:25]: Can `variableTypeIds` here be declared as `VariableType[]` (noting that VariableType here was the renaming of VariableTypeEnum)

File: src/main/jhipster/src/main/webapp/app/shared/ontology/data-type.enum.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export enum DataTypeEnum {
```
Comment (darla-leafnode) [inline src/main/jhipster/src/main/webapp/app/shared/ontology/data-type.enum.ts:1]: Just rename as `DataType` (omit enum)

File: src/main/jhipster/src/main/webapp/app/shared/germplasm/service/germplasm.service.ts [hunk 7]
```diff
@@ -71,20 +71,8 @@ export class GermplasmService {
...
-
-    getGermplasmAttributesByType(attributeType: string): Observable<Attribute[]> {
-        const url = SERVER_API_URL + `crops/${this.context.cropName}/germplasm/attributes` +
-            '?programUUID=' + this.context.programUUID + '&type=' + attributeType;
-        return this.http.get<Attribute[]>(url);
-    }
-
-    getGermplasmAttributesByGidAndType(gid: number, type: string): Observable<GermplasmAttribute[]> {
-        const url = SERVER_API_URL + `crops/${this.context.cropName}/germplasm/${gid}/attributes?type=${type}`;
+    getGermplasmAttributesByGidAndType(gid: number, variableTypeId: number): Observable<GermplasmAttribute[]> {
```
Comment (darla-leafnode) [inline src/main/jhipster/src/main/webapp/app/shared/germplasm/service/germplasm.service.ts:74]: Can `variableTypeId `here be declard as `VariableType` (suggested to be renamed from VariableTypeEnum currently)?

**Pull request merged**